### PR TITLE
[ticket/17130] Change storage of enable_magic_url in Posts and PMs

### PR DESCRIPTION
**Pull request by iMattPro**
Checklist:

- [x] Correct branch: master for new features; 3.3.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/master/coding-guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-17130

I don't know if we also need a migration to update these fields in post and PM tables, at least for boards that have Allow magic URLs disabled board wide. I've got a migration locked and loaded but have not put it here yet.

### CODE REVIEW

File: phpBB/includes/ucp/ucp_pm_compose.php [hunk 1]
```diff
@@ -828,7 +828,7 @@ function compose_pm($id, $mode, $action, $user_folders = array())
 
 		$enable_bbcode 		= (!$bbcode_status || isset($_POST['disable_bbcode'])) ? false : true;
 		$enable_smilies		= (!$smilies_status || isset($_POST['disable_smilies'])) ? false : true;
-		$enable_urls 		= (isset($_POST['disable_magic_url'])) ? 0 : 1;
+		$enable_urls 		= (!$url_status || isset($_POST['disable_magic_url'])) ? 0 : 1;
```
Comment (rxu) [inline phpBB/includes/ucp/ucp_pm_compose.php:831]: Will it work like that?
```php
$enable_urls 		= ($url_status || !$request->is_set_post('disable_magic_url'));
```

Also, it probably requires `&&` as far as disallowing links in posts means disallowing them regardless of `'disable_magic_url'` option.

**Comment by iMattPro** [comment]
I put up the migration. What it would do, and this would only apply to boards that have disallowed board-wide Magic URLs in posts and PMs:
- First sets all `enable_magic_url` in posts and PMs to `0`
- Then start looking at all the posts and PMs for any that actually contain rendered Magic URLs and change those `enable_magic_url` fields back to `1`
This may be a bit of a big hammer. But in reality if a board has disallowed Magic URLs then that field should have always been set to 0 anyways, so this migration fixes that.

If the admin later decides to start allowing magic URL's again, this would be the behavior:
- New posts will have magic URLs
- Old posts still won't be magic.
- Running the text reporter on posts or PMs will not turn old posts into magic URLs, they will remain as before.
- Editing an old post, the checkbox for Do not parse URLs will be checked by default, but could be un-checked to make the URLs magic after editing.

File: phpBB/includes/ucp/ucp_pm_compose.php (hunk 1 above)
Comment (iMattPro) [reply phpBB/includes/ucp/ucp_pm_compose.php:831]: No it needs to be an OR actually, exactly like the smilies and bbcode ones above

**Comment by marc1706** [comment]
Closing in favor of #6604

**Pull request closed without merging**